overlord/snapstate/snapmgr.go

```diff
@@ -224,3 +212,248 @@ func (m *SnapManager) Wait() {
 func (m *SnapManager) Stop() {
 	m.runner.Stop()
 }
+
+// helper to find the path and developer from an installState
+// (it may either be a local snap or downloaded from the store)
+func snapPathAndDeveloperFromInstState(t *state.Task, inst *installState) (string, string, error) {
+	if inst.SnapPath != "" {
+		return inst.SnapPath, snappy.SideloadedDeveloper, nil
+	} else if inst.DownloadTaskID != "" {
+		var dl downloadState
+
+		t.State().Lock()
+		tDl := t.State().Task(inst.DownloadTaskID)
+		err := tDl.Get("download-state", &dl)
+		t.State().Unlock()
+		if err != nil {
+			return "", "", err
+		}
+		return dl.SnapPath, dl.Developer, nil
+	}
+
+	return "", "", fmt.Errorf("internal error: installState created without a snap path source")
+}
+
+func getSnapSetupState(t *state.Task, setup *setupState) error {
+	var id string
+	t.State().Lock()
+	err := t.Get("setup-snap-id", &id)
+	t.State().Unlock()
+	if err != nil {
+		return nil
+	}
+	t.State().Lock()
+	ts := t.State().Task(id)
+	err = ts.Get("setup-state", setup)
+	t.State().Unlock()
+
+	return err
+}
+
+func (m *SnapManager) doCheckSnap(t *state.Task, _ *tomb.Tomb) error {
+	var inst installState
+
+	t.State().Lock()
+	err := t.Get("install-state", &inst)
+	t.State().Unlock()
+	if err != nil {
+		return err
+	}
+
+	snapPath, _, err := snapPathAndDeveloperFromInstState(t, &inst)
+	if err != nil {
+		return err
+	}
+
+	return m.backend.CheckSnap(snapPath, inst.Flags)
+}
+
+func (m *SnapManager) undoMountSnap(t *state.Task, _ *tomb.Tomb) error {
+	var inst installState
+
+	t.State().Lock()
+	err := t.Get("install-state", &inst)
+	t.State().Unlock()
+	if err != nil {
+		return err
+	}
+
+	snapPath, _, err := snapPathAndDeveloperFromInstState(t, &inst)
+	if err != nil {
+		return err
+	}
+
+	return m.backend.UndoSetupSnap(snapPath)
+}
+
+func (m *SnapManager) doMountSnap(t *state.Task, _ *tomb.Tomb) error {
+	var inst installState
+
+	t.State().Lock()
+	err := t.Get("install-state", &inst)
```

Comment:
For the tasks we keep (considering comment above), let's please name their main state structure after the task name, even if it looks similar to installState today. This allows us to diverge the states without breaking compatibility with old code and making it look awkward (having an install-state in the mount task which is not installState).